src/ngx_http_upsync_module.c

```diff
@@ -461,22 +463,57 @@ ngx_http_upsync_server(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
 
     ngx_memzero(&u, sizeof(ngx_url_t));
 
-    p = (u_char *)ngx_strchr(value[1].data, '/');
+    p = (u_char *)ngx_strchr(value[1].data, '@');
+    if (p == NULL) {
+        s.data = value[1].data;
+        s.len = value[1].len;
+
+        upscf->upsync_basic_auth.len = 0;
+        upscf->upsync_basic_auth.data = ngx_pcalloc(cf->pool, 1);
+    }
+    else {
+        size_t      len;
+        ngx_str_t   encoded, decoded;
+        u_char      *h;
+
+        s.data = p + 1;
+        s.len = value[1].len - (p + 1 - value[1].data);
+
+        /* Authorization: Basic base64(userpass) */
+        decoded.len = p - value[1].data;
+        decoded.data = value[1].data;
+
+        len = sizeof("Authorization: Basic ") - 1 + ngx_base64_encoded_length(decoded.len) + sizeof("\r\n");
```

Comment:
Each line shoud not be more than 80 characters or too much, please start a new line.